docs/contributing.md

```diff
@@ -0,0 +1,159 @@
+Welcome to Casdoor! This document is a guideline about how to contribute to Casdoor.
+
+If you find something incorrect or missing, please leave comments / suggestions.
+
+## Get involved
+
+There are many ways to contribute to Casdoor, here are some ideas to get started:
+
+Use Casdoor and report issues! When using Casdoor, report issues to promote development of Casdoor, no matter bugs or 
+proposal. Before file an issue on GitHub, it would be better to discuss first on [Gitter](https://gitter.im/casbin/casdoor),
+[Casbin Forum](https://forum.casbin.com/) or QQ group: [645200447](https://qm.qq.com/cgi-bin/qm/qr?k=SCBnKNj_1ljeXFT2dk8cwoGQwc5lFy8l&jump_from=webapi)
+
+:::info
+
+When reporting an issue, please use English to describe the details of you problem.
+
+:::
+
+Help with docs! Contributing start from docs are a good choice to start your contribution.
+
+Help solve issues! We prepare a table containing easy tasks suitable for beginner, with different level of challenges 
+labeled with different tags, check the table here [Casdoor Easy Tasks](https://github.com/orgs/casdoor/projects/1).
+
+## Contributing
+
+Now, if you are ready to create PR, here is the workflow for contributors:
+
+1. Fork to your own
+
+2. Clone fork to a local repository
+
+3. Create a new branch and work on it
+
+4. Keep your branch in sync
+
+5. Commit your changes (make sure your commit message is concise)
+
+6. Push your commits to your forked repository
+
+7. Create a pull request to **master** branch.
+
+## Pull Requests
+
+### Before you get started
+
+Casdoor uses GitHub as its developing platform. So the pull requests are the main source of contributions.
+
+There are something you need to know before you open a pull request:
+
+- When you first pull request, you need to sign the **CLA**.
+
+- Explain why you send this PR and what this PR would do to the repo.
+
+- **One commit is allowed**. Make sure the PR does only one thing, otherwise please split it.
+
+- If there are newly added files, please include Casdoor license to the top of new file(s).
+
+```text
+// Copyright 2022 The Casdoor Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+```
+
+### Semantic PRs
+
+Your pull requests should follow the Conventional Commits spec. The basic requirement is that only the PR title or at 
+least one commit message. For example, three commonly used PR title are given below:
+
+:::caution
+The PR title must be in lower case.
```

Comment:
```suggestion

The PR title must be in lower case.

```

Reply:
how to set the font red?

Reply:
refer to https://docusaurus.io/docs/markdown-features/admonitions

```mdx
:::danger

Some **content** with _Markdown_ `syntax`. Check [this `api`](#).

:::
```
<img width="719" alt="image" src="https://user-images.githubusercontent.com/53544726/182826272-26261f8f-47e0-4b80-a9ea-a8970d83278c.png">